crates/jxl-grid/src/simple_grid.rs

```diff
@@ -20,32 +20,58 @@ const fn compute_align<S>() -> usize {
 /// A continuous buffer in the "raster order".
 ///
 /// The buffer is aligned so that it can be used in SIMD instructions.
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub struct SimpleGrid<S> {
     width: usize,
     height: usize,
     offset: usize,
     buf: Vec<S>,
+    handle: Option<AllocHandle>,
 }
 
 impl<S: Default + Clone> SimpleGrid<S> {
     const ALIGN: usize = compute_align::<S>();
 
-    /// Create a new buffer.
+    /// Create a new buffer, recording the allocation if a tracker is given.
     #[inline]
-    pub fn new(width: usize, height: usize) -> Self {
+    pub fn with_alloc_tracker(
+        width: usize,
+        height: usize,
+        tracker: Option<&AllocTracker>,
+    ) -> Result<Self, Error> {
         let len = width * height;
-        let mut buf = vec![S::default(); len + Self::ALIGN / std::mem::size_of::<S>()];
+        let buf_len = len + (Self::ALIGN - 1) / std::mem::size_of::<S>();
```

Comment:
Just to be sure, is change to `(Self::ALIGN - 1)` intended?

Reply:
Yep, because `((Self::ALIGN - extra) % Self::ALIGN) <= Self::ALIGN - 1`. I noticed that it was allocating more space than needed.